frontera/tests/test_frontera_scheduler.py

```diff
@@ -0,0 +1,150 @@
+        assert all([isinstance(seed, FRequest) for seed in fs.frontier.manager.seeds])
+        assert fs.stats_manager.stats.get_value('frontera/seeds_count') == 3
+
+    def test_redirect_disabled_enqueue_requests(self):
+        settings = Settings()
+        settings['REDIRECT_ENABLED'] = False
+        crawler = FakeCrawler(settings)
+        fs = FronteraScheduler(crawler, manager=FakeFrontierManager)
+        fs.open(Spider)
+        assert fs.enqueue_request(rr1) is False
+        assert fs.enqueue_request(rr2) is False
+        assert fs.enqueue_request(rr3) is True
+        assert isinstance(fs.frontier.manager.seeds[0], FRequest)
+        assert len(fs.frontier.manager.seeds) == 1
+        assert fs.frontier.manager.seeds[0].url == rr3.url
+        assert fs.stats_manager.stats.get_value('frontera/seeds_count') == 1
+
+    def test_redirect_enabled_enqueue_requests(self):
+        settings = Settings()
+        settings['REDIRECT_ENABLED'] = True
+        crawler = FakeCrawler(settings)
+        fs = FronteraScheduler(crawler, manager=FakeFrontierManager)
+        fs.open(Spider)
+        assert fs.enqueue_request(rr1) is True
+        assert fs.enqueue_request(rr2) is True
+        assert fs.enqueue_request(rr3) is True
+        assert len(fs.frontier.manager.seeds) == 1
+        assert isinstance(fs.frontier.manager.seeds[0], FRequest)
+        assert fs.frontier.manager.seeds[0].url == rr3.url
+        assert set([request.url for request in fs._pending_requests]) == set([rr1.url, rr2.url])
+        assert all([isinstance(request, Request) for request in fs._pending_requests])
+        assert fs.stats_manager.stats.get_value('frontera/seeds_count') == 1
+        assert fs.stats_manager.stats.get_value('frontera/redirected_requests_count') == 2
+
+    def test_next_request(self):
+        crawler = FakeCrawler()
+        fs = FronteraScheduler(crawler, manager=FakeFrontierManager)
+        fs.open(Spider)
+        fs.frontier.manager.put_requests([fr1, fr2, fr3])
+        requests = [fs.next_request() for _ in range(3)]
+        assert set([request.url for request in requests]) == set([fr1.url, fr2.url, fr3.url])
+        assert all([isinstance(request, Request) for request in requests])
+        assert fs.stats_manager.stats.get_value('frontera/returned_requests_count') == 3
+
+    def test_next_request_manager_finished(self):
+        crawler = FakeCrawler()
+        fs = FronteraScheduler(crawler, manager=FakeFrontierManager)
+        fs.open(Spider)
+        fs.frontier.manager.put_requests([fr1])
+        fs.frontier.manager.finished = True
+        assert fs.next_request() is None
+        assert fs.stats_manager.stats.get_value('frontera/returned_requests_count') is None
+
+    def test_next_request_overused_keys_info(self):
+        settings = Settings()
+        settings['CONCURRENT_REQUESTS_PER_DOMAIN'] = 0
+        settings['CONCURRENT_REQUESTS_PER_IP'] = 1
```

Comment:
I have made this change, although since we are using a fake downloader, other changes are not needed in the test to make it work.